client/asset/btc/wallettypes.go

```diff
@@ -95,3 +95,20 @@ type RPCOutpoint struct {
 	TxID string `json:"txid"`
 	Vout uint32 `json:"vout"`
 }
+
+// GetWalletInfoResult models the data from the getwalletinfo command.
+type GetWalletInfoResult struct {
+	WalletName            string  `json:"walletname"`
+	WalletVersion         int     `json:"walletversion"`
+	Balance               float64 `json:"balance"`
+	UnconfirmedBalance    float64 `json:"unconfirmed_balance"`
+	ImmatureBalance       float64 `json:"immature_balance"`
+	TxCount               int     `json:"txcount"`
+	KeyPoolOldest         uint64  `json:"keypoololdest"`
+	KeyPoolSize           int     `json:"keypoolsize"`
+	KeyPoolSizeHDInternal int     `json:"keypoolsize_hd_internal"`
+	PayTxFee              float64 `json:"paytxfee"`
+	HdSeedID              string  `json:"hdseedid"`
+	HdMasterKeyID         string  `json:"hdmasterkeyid"`
+	PriveyKeysEnabled     bool    `json:"private_keys_enabled"`
+}
```

Comment:
Since we might want to use this RPC for purposes other than balance with >0.18 nodes, which has apparently added some fields (at least `avoid_reuse` and a `scanning` object), I think we should consider calling this `GetWalletInfoLegacy` or similar plus similar adjustments to the method that uses this.

For LTC 0.18.1, I don't see `"hdmasterkeyid"` in the result with `help getwalletinfo`.

LTC 0.18.1 does have a `unlocked_until` field that is presumably only included for locked wallets that might be useful so we might as well unmarshal it if present.

Alt we could make a struct that works for both new and old with the union of the fields between the versions and appropriate use of omitempty.

Reply:
Also, I'm curious about the choice of integer types here (`int` vs `uint64`).  Is there ref or specific reasoning?

Reply:
`keypoololdest` is a timestamp. `txcount` and `keypoolsize` are both counts that should be safe as 32-bit signed int. Version is a 6-digit number, so also safe. Would you feel better with `uint32`'s?

> Alt we could make a struct that works for both new and old with the union of the fields between the versions

^^^ this please. At least until we come across a field that is needed and not shared.

Reply:
Just that `int` is a 32-bit int on 32-bit systems and 64-bit on 64-bit systems, but the spec only guarantees that they are 32 bits on any platform.  I just figure if we want 32 bit we should specify it explicitly.